src/System.Private.ServiceModel/tools/SelfHostedWcfService/Program.cs

```diff
@@ -274,6 +266,15 @@ static void Main()
             WebSocketHttpsDuplexTextBufferedTestServiceHost webSocketHttpsDuplexTextBufferedTestServiceHostServiceHost = new WebSocketHttpsDuplexTextBufferedTestServiceHost(typeof(WcfService.WSDuplexService), webSocketHttpsDuplexTextBufferedTestServiceHostbaseAddress);
             webSocketHttpsDuplexTextBufferedTestServiceHostServiceHost.Open();
```

Comment:
Suggestion only: I almost feel like all these Uri/ServiceHost pairs should go into a Tuple<Uri, ServiceHost> pair and then you can loop over them to open.